R/protein_abundance_normalisation.R

```diff
@@ -0,0 +1,262 @@
+#' Perform normalization of LiP peptide data by protein abundance from tryptic controls
+#'
+#' LiP peptide intensities are normalised by protein abundances from tryptic controls to correct for protein fold changes due to treatment. 
+#'
+#' @param lip_peptides A data frame containing LiP peptide level data. The \code{sample, protein_id, grouping, condition, peptide_intensity} variables 
+#' are required. Ideally the output of \code{assign_missingness} or \code{impute} is used for the method \code{"ref_vs_rest"}, since here also the variable 
+#' \code{missingness} is required in contrast to method \code{"all_vs_all"}.
+#' @param tc_proteins A data frame containing tryptic control protein level data. The \code{sample, protein_id, condition, protein_intensity} variables
+#' are required. Ideally the output of \code{assign_missingness} or \code{impute} is used for the method \code{"ref_vs_rest"}, since here also the variable 
+#' \code{missingness} is required in contrast to method \code{"all_vs_all"}.
+#' @param sample The column in the data frame containing the sample name. Is not required if \code{method = "t-test_mean_sd"}.
+#' @param protein_id The column in the data frame containing the protein accession numbers. 
+#' @param grouping The column in the data frame containing precursor or peptide identifiers.
+#' @param condition The column in the data frame containing the conditions.
+#' @param peptide_intensity The column in the data frame containing peptide intensity values. The intensity values need to be log2 transformed.
+#' @param protein_intensity The column in the data frame containing protein intensity values. The intensity values need to be log2 transformed.
+#' @param method A character vector specifying which method should be used. If \code{method = "all_vs_all"} an ANOVA based method will be performed that 
+#' compares all conditions with each other followed by Tukey's honestly significant difference test to identify pairs of conditions that are significantly different.
+#' If \code{mehtod = "ref_vs_rest"} a reference condition will be compared against all other conditions using a Welch t-test in order to identify significantly
+#' changing peptides or precursors.
+#' @param paired A logical indicating if LiP and tryptic control data are paired. In other words it indicates if replicates of LiP and tryptic control belong together.
```

Comment:
I don't really understand what you mean here. 
Do you mean e.g. LiP 0 and TC 0 would be paired whereas LiP 0 and TC 5 would not be paired? Wouldn't you always use paired conditions?

Reply:
The paired parameter defines if replicate 1 of a condition in LiP belongs together to replicate 1 of the same condition in the tryptic control. If that is the case only protein abundances in the tryptic control are kept for which there is also a peptide measurement in LiP. 
Cathy used this because for her replicates in tryptic control and in LiP were paired. For me this is never the case so I would never use paired. It depends on your experimental setup.

Reply:
Ah makes sense. You might want to explain this a bit more clearly, otherwise ppl might get confused (like me)